src/graph/components/runtime.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from "react";
+import { IAuthoredState } from "./types";
+import {
+  addLinkedInteractiveStateListener, IInteractiveStateWithDataset, removeLinkedInteractiveStateListener, IDataset
+} from "@concord-consortium/lara-interactive-api";
+import { Bar } from "react-chartjs-2";
+import { generateChartData, emptyChartData } from "../generate-chart-data";
+
+export interface IProps {
+  authoredState: IAuthoredState;
+}
+
+const defaultGraphOptions = {
+  scales: {
+    yAxes: [
+      {
+        ticks: {
+          beginAtZero: true,
+        },
+      },
+    ],
+  },
+};
+
+const emptyGraphOptions = {
+  ...defaultGraphOptions,
+  legend: {
+    display: false
+  }
+};
+
+export const Runtime: React.FC<IProps> = ({ authoredState }) => {
+  const [ datasets, setDatasets ] = useState<Array<IDataset | null | undefined>>([]);
+
+  useEffect(() => {
+    const linkedInteractives: string[] = [
+      authoredState.dataSourceInteractive1,
+      authoredState.dataSourceInteractive2,
+      authoredState.dataSourceInteractive3
+    ].filter(intItemId => intItemId !== undefined) as string[];
+
+    const unsubscribeMethods = linkedInteractives.map((dataSourceInteractive, datasetIdx) => {
+      const listener = (newLinkedIntState: IInteractiveStateWithDataset | undefined) => {
+        const newDataset = newLinkedIntState && newLinkedIntState.dataset;
+        // Accept null or undefined datasets too to clear them. If it's an object, make sure it follows
+        // specified format.
+        if (!newDataset || newDataset && newDataset.type === "dataset" && Number(newDataset.version) === 1) {
+          setDatasets(prevDatasets => {
+            const newDatasets = prevDatasets.slice();
+            newDatasets[datasetIdx] = newDataset;
+            return newDatasets;
+          });
+        } else if (newDataset && newDataset.type === "dataset" && Number(newDataset.version) !== 1) {
```

Comment:
Repeating the expected version number in several places in these tests seems sub-optimal. Perhaps consolidate the test:
```typescript
  const isValidDatasetVersion = Number(newDataset.version) === 1;
```

Reply:
Done.